dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1797,5 +1800,94 @@ private void checkBatchOperateResult(String srcProjectName, String targetProject
         }
     }
 
+    public Map<String, Object> queryUpstreamTaskDependencies(Integer processId, String taskName) {
+        Map<String, Object> result = new HashMap<>();
+        List<String> depList = new ArrayList<>();
+
+        ProcessDefinition processDefinition = processDefineMapper.selectById(processId);
+        if (processDefinition == null) {
+            logger.info("Upstream dependencies process define not exists");
+            putMsg(result, Status.PROCESS_DEFINE_NOT_EXIST, processId);
+            return result;
+        }
+
+        String processDefinitionJson = processDefinition.getProcessDefinitionJson();
+
+        ProcessData processData = JSONUtils.parseObject(processDefinitionJson, ProcessData.class);
+
+        //process data check
+        if (null == processData) {
+            logger.error("Upstream dependencies process data is null");
+            putMsg(result, Status.DATA_IS_NOT_VALID, processDefinitionJson);
+            return result;
+        }
+
+        List<TaskNode> taskNodeList = (processData.getTasks() == null) ? new ArrayList<>() : processData.getTasks();
+
+        for (TaskNode taskNode : taskNodeList) {
+            if (taskNode.getName().equals(taskName)) {
+                //getPreTasks() holds inner dependency information
+                depList.add(taskNode.getPreTasks());
+            }
+
+        }
+
```

Comment:
Hi,if the pretasks contains dependence task,you should parse the dependene json.

Reply:
You can also abstract a class to describe the dependent item,it need includes common attributes：project id,process definition id,task name